datafusion/src/physical_plan/planner.rs

```diff
@@ -265,8 +266,46 @@ impl DefaultPhysicalPlanner {
                 }
 
                 let input_exec = self.create_initial_plan(input, ctx_state)?;
-                let physical_input_schema = input_exec.schema();
-                let logical_input_schema = input.as_ref().schema();
+
+                // at this moment we are guaranteed by the logical planner
+                // to have all the window_expr to have equal sort key
+                let sort_keys = match window_expr[0] {
+                    Expr::WindowFunction {
+                        ref partition_by,
+                        ref order_by,
+                        ..
+                    } => generate_sort_key(partition_by, order_by),
+                    _ => unreachable!(),
+                };
+
+                let input_exec = if sort_keys.is_empty() {
+                    input_exec
+                } else {
+                    let sort_keys = sort_keys
+                        .iter()
+                        .map(|e| match e {
+                            Expr::Sort {
+                                expr,
+                                asc,
+                                nulls_first,
+                            } => self.create_physical_sort_expr(
+                                expr,
+                                input.schema(),
+                                &input_exec.schema(),
```

Comment:
same here, we could avoid `schema()` call by reusing `&physical_input_schema`.

Reply:
fixed